File: fixtures/scala/src/main/scala/calculator/Calculator.scala
package calculator

class Calculator {
  private var lastResult: Double = 0.0

  def add(a: Double, b: Double): Double = {
    lastResult = Operations.add(a, b)
    lastResult
  }

  def subtract(a: Double, b: Double): Double = {
    lastResult = Operations.subtract(a, b)
    lastResult
  }

  def multiply(a: Double, b: Double): Double = {
    lastResult = Operations.multiply(a, b)
    lastResult
  }

  def divide(a: Double, b: Double): Double = {
    lastResult = Operations.divide(a, b)
    lastResult
  }

  def getLastResult: Double = lastResult
}

File: fixtures/java/src/main/java/calculator/Calculator.java
package calculator;

public class Calculator {
    private double lastResult;

    public Calculator() {
        this.lastResult = 0.0;
    }

    public double add(double a, double b) {
        lastResult = Operations.add(a, b);
        return lastResult;
    }

    public double subtract(double a, double b) {
        lastResult = Operations.subtract(a, b);
        return lastResult;
    }

    public double multiply(double a, double b) {
        lastResult = Operations.multiply(a, b);
        return lastResult;
    }

    public double divide(double a, double b) {
        lastResult = Operations.divide(a, b);
        return lastResult;
    }

    public double getLastResult() {
        return lastResult;
    }
}
